wpiutil/src/main/native/cpp/FileLogger.cpp

```diff
@@ -0,0 +1,35 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+
+#include "wpi/FileLogger.h"
+
+namespace wpi {
+FileLogger::FileLogger(std::string_view file,
+                       std::function<void(std::string_view)> callback)
+#ifdef __linux__
+    : m_fileHandle{open(file.data(), O_RDONLY)},
+      m_inotifyHandle{inotify_init()},
+      m_inotifyWatchHandle{
+          inotify_add_watch(m_inotifyHandle, file.data(), IN_MODIFY)},
+      m_thread{[=, this] {
+        char buf[4000];
+        struct inotify_event ev;
+        int len = 0;
+        while ((len = read(m_inotifyHandle, &ev, sizeof(ev))) > 0) {
+          int buf_len = read(m_fileHandle, buf, sizeof(buf));
+          callback(std::string_view{buf, buf_len});
```

Comment:
This should handle [`read()`](https://man7.org/linux/man-pages/man2/read.2.html) returning -1 to indicate an error and convert the result to `size_t` before passing it to the `std::string_view` constructor.